index.html

```diff
@@ -4116,91 +4060,475 @@ <h3>Parsing content</h3>
+          <a>NFC adapter</a> are read by unauthorized parties.
+        </dd>
+        <dt><strong>Affected assets</strong></dt>
+        <dd>
+          Confidential NDEF message payload in-transfer and in-storage.
+        </dd>
+        <dt><strong>Actors</strong></dt>
+        <dd>
+          Malicious MitM user, malicious web page creator.
+        </dd>
+        <dt><strong>Mitigation, comments</strong></dt>
+        <dd>
+          To ensure confidentiality, use payload encryption and secure
+          communication for data exchange, authentication and authorization
+          between Web NFC and <a>NFC adapter</a>s.
+        </dd>
+      </dl>
+    </section>
+
+    <section> <h4>Active attack via malicious NFC tag</h4>
+      <dl>
+        <dt><strong>Threat description</strong></dt>
+        <dd>
+          Malicious tag may be involuntarily or voluntarily read by devices
+          and the data read may constitute an attack vector on the user agent.
+          For example it can attempt to trigger an action on
+          the device which may be a threat, for instance launching a malicious
+          web site, or opening an image prepared for attacking the device.
+        </dd>
+        <dt><strong>Affected assets</strong></dt>
+        <dd>
+          Integrity of user device, all other Web NFC assets.
+        </dd>
+        <dt><strong>Actors</strong></dt>
+        <dd>
+          Malicious tag creator.
+        </dd>
+        <dt><strong>Mitigation, comments</strong></dt>
+        <dd>
+          This is a generic problem with all existing NFC tags.
+          The data is considered application specific.
+          Implementations need security hardening.
+          Involuntary touch is low probability due to short range and critical
+          angle for reading, and due to the focus requirements.
+          Automatic actions for smart posters and other tags should not be
+          allowed. The user must be made aware and given the ability to control
+          what is happening during the NFC communication. For instance, opening
+          content from <a>smart poster</a>, automatic connection to (possibly
+          malicious) WiFi via <a>NFC handover</a>, etc.
+          Do no allow actions from untrusted NFC tags, trust can be established
+          via the <a>NDEF signature</a> check.
+        </dd>
+      </dl>
+    </section>
+  </section>
+
+  <section class="informative"> <h2>Security mechanisms</h2>
+    <p>
+      In order to support security policies concerning Web NFC, the following
+      mechanisms are recommended for the implementations.
+    </p>
+    <section> <h3><dfn>Obtaining permission</dfn></h3>
+      Implementations SHOULD use one or more of the following mechanisms to
+      <dfn>obtain permission</dfn>.
+      <ul>
+        <li>
+          A mechanism for obtaining <a>expressed permission</a> of the user.
+        </li>
+        <li>
+          A mechanism for <a>ask for forgiveness</a> policies.
```

Comment:
Shall we be more conservative and add "ask for forgiveness" to the spec when we actually witness user agents that feel the need of having this kind of mechanism?

Reply:
Or we change "recommended" mechanism to "possible"?

Or should I plainly remove (comment out) that mechanism?

Reply:
I would simply remove it that mechanism. What do you think @kenchris?